test/csuite/wt8659_reconstruct_database_from_logs/main.c

```diff
@@ -0,0 +1,330 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <test_util.h>
+
+#define MAX_ITERATIONS 5
+#define MAX_KEYS 100000
+
+static const char *const conn_config =
+  "create,cache_size=100MB,log=(archive=false,enabled=true,file_max=100K)";
+static const char *const home_live = "WT_HOME_LOG";
+static const char *const home_full = "WT_HOME_LOG_FULL";
+static const char *const home_incr = "WT_HOME_LOG_INCR";
+static const char *const home_incr_copy = "WT_HOME_LOG_INCR_COPY";
+static const char *const uri = "table:logtest";
+static const char *const full_out = "./backup_full";
+static const char *const incr_out = "./backup_incr";
+
+static WT_CONNECTION *conn = NULL;
+static WT_SESSION *session = NULL;
+
+/*
+ * dump_table --
+ *     Dumps the table content in to the file in human-readable format.
+ */
+static void
+dump_table(const char *home, const char *table, const char *out_file)
+{
+    char buf[1024];
+
+    testutil_check(
+      __wt_snprintf(buf, sizeof(buf), "./wt -R -h %s dump %s > %s", home, table, out_file));
+    testutil_check(system(buf));
+}
+
+/*
+ * reset_dir --
+ *     Recreates the directory.
+ */
+static void
+reset_dir(const char *dir)
+{
+    char buf[1024];
+
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "rm -rf %s && mkdir %s", dir, dir));
```

Comment:
np: Can't we just delete the content of the directory?

Reply:
By removing the entire directory and recreating it, it allows for the case of the directory not existing at all or if it exists. If you only remove the content, then you build in an assumption that the directory must exist before calling. This way this function can be used anywhere, anytime in the program.

Reply:
@sueloverso, thanks for the explanation!